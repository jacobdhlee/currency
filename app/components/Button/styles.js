import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({
  container: {
    alignItems: 'center',
  },
  wrap: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  text: {
    color: '$white',
    fontWeight: '300',
    fontSize: 15,
    paddingVertical: 20,
  },
  icon: {
    width: 19,
    marginRight: 11,
  },
});
