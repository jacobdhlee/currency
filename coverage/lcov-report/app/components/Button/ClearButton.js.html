<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/Button/ClearButton.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">app/components/Button</a> ClearButton.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/17</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/8</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js"><span class="cstat-no" title="statement not covered" >import React from 'react';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span></span>
import PropType<span class="cstat-no" title="statement not covered" >s from 'prop-types';<span class="cstat-no" title="statement not covered" ></span></span>
import style<span class="cstat-no" title="statement not covered" >s from './styles';<span class="cstat-no" title="statement not covered" ></span></span>
import ima<span class="cstat-no" title="statement not covered" >ge from './images/icon.png';<span class="cstat-no" title="statement not covered" ></span></span>
import { View, Te<span class="cstat-no" title="statement not covered" >xt, TouchableOpacity, Im</span>age } fro<span class="fstat-no" title="function not covered" >m 'react-native';<span class="cstat-no" title="statement not covered" ></span></span>
&nbsp;
const ClearButto<span class="cstat-no" title="statement not covered" >n = ({ on<span class="fstat-no" title="function not covered" >Press, text </span>}) =&gt; (<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span></span></span>
  &lt;TouchableOpacity style={styles.container} onPress={onPress}&gt;
    &lt;View style={styles.wrap}&gt;
      &lt;Image resizeMode="contain" style={styles.icon} source={image} /&gt;
      &lt;Text style={styles.text}&gt; {text} &lt;/Text&gt;
    &lt;/View&gt;
  &lt;/TouchableOpacity&gt;
);
&nbsp;
<span class="cstat-no" title="statement not covered" >ClearButton.prototype = {</span>
  onPress: PropTypes.func,
  text: PropTypes.string,<span class="cstat-no" title="statement not covered" ></span>
};
&nbsp;
export default ClearButton;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Dec 16 2017 14:22:19 GMT-0500 (EST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
