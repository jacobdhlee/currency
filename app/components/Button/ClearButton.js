import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles';
import image from './images/icon.png';
import { View, Text, TouchableOpacity, Image } from 'react-native';

const ClearButton = ({ onPress, text }) => (
  <TouchableOpacity style={styles.container} onPress={onPress}>
    <View style={styles.wrap}>
      <Image resizeMode="contain" style={styles.icon} source={image} />
      <Text style={styles.text}> {text} </Text>
    </View>
  </TouchableOpacity>
);

ClearButton.prototype = {
  onPress: PropTypes.func,
  text: PropTypes.string,
};

export default ClearButton;
