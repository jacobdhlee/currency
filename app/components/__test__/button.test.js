import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'react-dom';
import { TouchableOpacity } from 'react-native';
import sinon from 'sinon';
import buildStyles from '../../config/styles';
import { ClearButton, styles } from '../Button';

beforeAll(() => {
  buildStyles();
});

it('render successfully', () => {
  const rendered = renderer.create(<ClearButton />).toJSON();
  expect(rendered).toBeTruthy();
});

it('styles', () => {
  expect(typeof styles).toBe('object');
});

it('text props passed', () => {
  const rendered = renderer
    .create(<ClearButton text="this is test" />)
    .toJSON();
  expect(rendered).toMatchSnapshot();
});

it('handle a press', () => {
  const callback = sinon.spy();
  const wrapper = shallow(<ClearButton text="this should handle a press" onPress={callback} />);
  expect(wrapper.find(TouchableOpacity).length).toBe(1);

  wrapper.find(TouchableOpacity).first().props().onPress();

  expect(callback.called).toBe(true);
});
