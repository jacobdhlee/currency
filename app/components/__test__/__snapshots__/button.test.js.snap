// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text props passed 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "marginRight": 11,
          "width": 19,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 15,
          "fontWeight": "300",
          "paddingVertical": 20,
        }
      }
    >
       
      this is test
       
    </Text>
  </View>
</View>
`;
